plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -322,10 +322,9 @@ else if (type instanceof ScaledFloatType) {
             return new TypeAndDecoder(DOUBLE, new DoubleDecoder.Descriptor(path));
         }
         else if (type instanceof DateTimeType dateTimeType) {
-            if (dateTimeType.getFormats().isEmpty()) {
-                return new TypeAndDecoder(TIMESTAMP_MILLIS, new TimestampDecoder.Descriptor(path));
-            }
-            // otherwise, skip -- we don't support custom formats, yet
+            int precision = dateTimeType.getPrecision();
```

Comment:
is precision always present or can it be missing?

Reply:
Currently the precision can only be 3 or 9, it will not be absent.